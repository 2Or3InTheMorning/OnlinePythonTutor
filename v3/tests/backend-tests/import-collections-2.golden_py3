{
  "code": "from collections import defaultdict\nd = defaultdict(list)\nd['hello'].append('a')\nd['hello'].append('b')\nd['hello'].append('c')\n", 
  "trace": [
    {
      "ordered_globals": [], 
      "stdout": "", 
      "func_name": "<module>", 
      "stack_to_render": [], 
      "globals": {}, 
      "heap": {}, 
      "line": 1, 
      "event": "step_line"
    }, 
    {
      "ordered_globals": [
        "defaultdict"
      ], 
      "stdout": "", 
      "func_name": "<module>", 
      "stack_to_render": [], 
      "globals": {
        "defaultdict": [
          "REF", 
          1
        ]
      }, 
      "heap": {
        "1": [
          "CLASS", 
          "defaultdict", 
          [
            "dict"
          ], 
          [
            "__copy__", 
            [
              "REF", 
              2
            ]
          ], 
          [
            "__getattribute__", 
            [
              "REF", 
              3
            ]
          ], 
          [
            "__init__", 
            [
              "REF", 
              4
            ]
          ], 
          [
            "__missing__", 
            [
              "REF", 
              5
            ]
          ], 
          [
            "__reduce__", 
            [
              "REF", 
              6
            ]
          ], 
          [
            "__repr__", 
            [
              "REF", 
              7
            ]
          ], 
          [
            "copy", 
            [
              "REF", 
              8
            ]
          ], 
          [
            "default_factory", 
            [
              "REF", 
              9
            ]
          ]
        ], 
        "2": [
          "INSTANCE", 
          "method_descriptor"
        ], 
        "3": [
          "INSTANCE", 
          "wrapper_descriptor"
        ], 
        "4": [
          "INSTANCE", 
          "wrapper_descriptor"
        ], 
        "5": [
          "INSTANCE", 
          "method_descriptor"
        ], 
        "6": [
          "INSTANCE", 
          "method_descriptor"
        ], 
        "7": [
          "INSTANCE", 
          "wrapper_descriptor"
        ], 
        "8": [
          "INSTANCE", 
          "method_descriptor"
        ], 
        "9": [
          "INSTANCE", 
          "member_descriptor"
        ]
      }, 
      "line": 2, 
      "event": "step_line"
    }, 
    {
      "ordered_globals": [
        "defaultdict", 
        "d"
      ], 
      "stdout": "", 
      "func_name": "<module>", 
      "stack_to_render": [], 
      "globals": {
        "defaultdict": [
          "REF", 
          1
        ], 
        "d": [
          "REF", 
          10
        ]
      }, 
      "heap": {
        "1": [
          "CLASS", 
          "defaultdict", 
          [
            "dict"
          ], 
          [
            "__copy__", 
            [
              "REF", 
              2
            ]
          ], 
          [
            "__getattribute__", 
            [
              "REF", 
              3
            ]
          ], 
          [
            "__init__", 
            [
              "REF", 
              4
            ]
          ], 
          [
            "__missing__", 
            [
              "REF", 
              5
            ]
          ], 
          [
            "__reduce__", 
            [
              "REF", 
              6
            ]
          ], 
          [
            "__repr__", 
            [
              "REF", 
              7
            ]
          ], 
          [
            "copy", 
            [
              "REF", 
              8
            ]
          ], 
          [
            "default_factory", 
            [
              "REF", 
              9
            ]
          ]
        ], 
        "2": [
          "INSTANCE", 
          "method_descriptor"
        ], 
        "3": [
          "INSTANCE", 
          "wrapper_descriptor"
        ], 
        "4": [
          "INSTANCE", 
          "wrapper_descriptor"
        ], 
        "5": [
          "INSTANCE", 
          "method_descriptor"
        ], 
        "6": [
          "INSTANCE", 
          "method_descriptor"
        ], 
        "7": [
          "INSTANCE", 
          "wrapper_descriptor"
        ], 
        "8": [
          "INSTANCE", 
          "method_descriptor"
        ], 
        "9": [
          "INSTANCE", 
          "member_descriptor"
        ], 
        "10": [
          "INSTANCE", 
          "defaultdict"
        ]
      }, 
      "line": 3, 
      "event": "step_line"
    }, 
    {
      "ordered_globals": [
        "defaultdict", 
        "d"
      ], 
      "stdout": "", 
      "func_name": "<module>", 
      "stack_to_render": [], 
      "globals": {
        "defaultdict": [
          "REF", 
          1
        ], 
        "d": [
          "REF", 
          10
        ]
      }, 
      "heap": {
        "1": [
          "CLASS", 
          "defaultdict", 
          [
            "dict"
          ], 
          [
            "__copy__", 
            [
              "REF", 
              2
            ]
          ], 
          [
            "__getattribute__", 
            [
              "REF", 
              3
            ]
          ], 
          [
            "__init__", 
            [
              "REF", 
              4
            ]
          ], 
          [
            "__missing__", 
            [
              "REF", 
              5
            ]
          ], 
          [
            "__reduce__", 
            [
              "REF", 
              6
            ]
          ], 
          [
            "__repr__", 
            [
              "REF", 
              7
            ]
          ], 
          [
            "copy", 
            [
              "REF", 
              8
            ]
          ], 
          [
            "default_factory", 
            [
              "REF", 
              9
            ]
          ]
        ], 
        "2": [
          "INSTANCE", 
          "method_descriptor"
        ], 
        "3": [
          "INSTANCE", 
          "wrapper_descriptor"
        ], 
        "4": [
          "INSTANCE", 
          "wrapper_descriptor"
        ], 
        "5": [
          "INSTANCE", 
          "method_descriptor"
        ], 
        "6": [
          "INSTANCE", 
          "method_descriptor"
        ], 
        "7": [
          "INSTANCE", 
          "wrapper_descriptor"
        ], 
        "8": [
          "INSTANCE", 
          "method_descriptor"
        ], 
        "9": [
          "INSTANCE", 
          "member_descriptor"
        ], 
        "10": [
          "INSTANCE", 
          "defaultdict"
        ]
      }, 
      "line": 4, 
      "event": "step_line"
    }, 
    {
      "ordered_globals": [
        "defaultdict", 
        "d"
      ], 
      "stdout": "", 
      "func_name": "<module>", 
      "stack_to_render": [], 
      "globals": {
        "defaultdict": [
          "REF", 
          1
        ], 
        "d": [
          "REF", 
          10
        ]
      }, 
      "heap": {
        "1": [
          "CLASS", 
          "defaultdict", 
          [
            "dict"
          ], 
          [
            "__copy__", 
            [
              "REF", 
              2
            ]
          ], 
          [
            "__getattribute__", 
            [
              "REF", 
              3
            ]
          ], 
          [
            "__init__", 
            [
              "REF", 
              4
            ]
          ], 
          [
            "__missing__", 
            [
              "REF", 
              5
            ]
          ], 
          [
            "__reduce__", 
            [
              "REF", 
              6
            ]
          ], 
          [
            "__repr__", 
            [
              "REF", 
              7
            ]
          ], 
          [
            "copy", 
            [
              "REF", 
              8
            ]
          ], 
          [
            "default_factory", 
            [
              "REF", 
              9
            ]
          ]
        ], 
        "2": [
          "INSTANCE", 
          "method_descriptor"
        ], 
        "3": [
          "INSTANCE", 
          "wrapper_descriptor"
        ], 
        "4": [
          "INSTANCE", 
          "wrapper_descriptor"
        ], 
        "5": [
          "INSTANCE", 
          "method_descriptor"
        ], 
        "6": [
          "INSTANCE", 
          "method_descriptor"
        ], 
        "7": [
          "INSTANCE", 
          "wrapper_descriptor"
        ], 
        "8": [
          "INSTANCE", 
          "method_descriptor"
        ], 
        "9": [
          "INSTANCE", 
          "member_descriptor"
        ], 
        "10": [
          "INSTANCE", 
          "defaultdict"
        ]
      }, 
      "line": 5, 
      "event": "step_line"
    }, 
    {
      "ordered_globals": [
        "defaultdict", 
        "d"
      ], 
      "stdout": "", 
      "func_name": "<module>", 
      "stack_to_render": [], 
      "globals": {
        "defaultdict": [
          "REF", 
          1
        ], 
        "d": [
          "REF", 
          10
        ]
      }, 
      "heap": {
        "1": [
          "CLASS", 
          "defaultdict", 
          [
            "dict"
          ], 
          [
            "__copy__", 
            [
              "REF", 
              2
            ]
          ], 
          [
            "__getattribute__", 
            [
              "REF", 
              3
            ]
          ], 
          [
            "__init__", 
            [
              "REF", 
              4
            ]
          ], 
          [
            "__missing__", 
            [
              "REF", 
              5
            ]
          ], 
          [
            "__reduce__", 
            [
              "REF", 
              6
            ]
          ], 
          [
            "__repr__", 
            [
              "REF", 
              7
            ]
          ], 
          [
            "copy", 
            [
              "REF", 
              8
            ]
          ], 
          [
            "default_factory", 
            [
              "REF", 
              9
            ]
          ]
        ], 
        "2": [
          "INSTANCE", 
          "method_descriptor"
        ], 
        "3": [
          "INSTANCE", 
          "wrapper_descriptor"
        ], 
        "4": [
          "INSTANCE", 
          "wrapper_descriptor"
        ], 
        "5": [
          "INSTANCE", 
          "method_descriptor"
        ], 
        "6": [
          "INSTANCE", 
          "method_descriptor"
        ], 
        "7": [
          "INSTANCE", 
          "wrapper_descriptor"
        ], 
        "8": [
          "INSTANCE", 
          "method_descriptor"
        ], 
        "9": [
          "INSTANCE", 
          "member_descriptor"
        ], 
        "10": [
          "INSTANCE", 
          "defaultdict"
        ]
      }, 
      "line": 5, 
      "event": "return"
    }
  ]
}
