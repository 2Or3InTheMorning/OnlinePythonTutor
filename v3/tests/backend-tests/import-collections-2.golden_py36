{
  "code": "from collections import defaultdict\nd = defaultdict(list)\nd['hello'].append('a')\nd['hello'].append('b')\nd['hello'].append('c')\n",
  "trace": [
    {
      "line": 1,
      "event": "step_line",
      "func_name": "<module>",
      "globals": {},
      "ordered_globals": [],
      "stack_to_render": [],
      "heap": {},
      "stdout": ""
    },
    {
      "line": 2,
      "event": "step_line",
      "func_name": "<module>",
      "globals": {
        "defaultdict": [
          "REF",
          1
        ]
      },
      "ordered_globals": [
        "defaultdict"
      ],
      "stack_to_render": [],
      "heap": {
        "1": [
          "CLASS",
          "defaultdict",
          [
            "dict"
          ],
          [
            "__copy__",
            [
              "REF",
              2
            ]
          ],
          [
            "__getattribute__",
            [
              "REF",
              3
            ]
          ],
          [
            "__init__",
            [
              "REF",
              4
            ]
          ],
          [
            "__missing__",
            [
              "REF",
              5
            ]
          ],
          [
            "__reduce__",
            [
              "REF",
              6
            ]
          ],
          [
            "__repr__",
            [
              "REF",
              7
            ]
          ],
          [
            "copy",
            [
              "REF",
              8
            ]
          ],
          [
            "default_factory",
            [
              "REF",
              9
            ]
          ]
        ],
        "2": [
          "INSTANCE",
          "method_descriptor"
        ],
        "3": [
          "INSTANCE",
          "wrapper_descriptor"
        ],
        "4": [
          "INSTANCE",
          "wrapper_descriptor"
        ],
        "5": [
          "INSTANCE",
          "method_descriptor"
        ],
        "6": [
          "INSTANCE",
          "method_descriptor"
        ],
        "7": [
          "INSTANCE",
          "wrapper_descriptor"
        ],
        "8": [
          "INSTANCE",
          "method_descriptor"
        ],
        "9": [
          "INSTANCE",
          "member_descriptor"
        ]
      },
      "stdout": ""
    },
    {
      "line": 3,
      "event": "step_line",
      "func_name": "<module>",
      "globals": {
        "defaultdict": [
          "REF",
          1
        ],
        "d": [
          "REF",
          10
        ]
      },
      "ordered_globals": [
        "defaultdict",
        "d"
      ],
      "stack_to_render": [],
      "heap": {
        "1": [
          "CLASS",
          "defaultdict",
          [
            "dict"
          ],
          [
            "__copy__",
            [
              "REF",
              2
            ]
          ],
          [
            "__getattribute__",
            [
              "REF",
              3
            ]
          ],
          [
            "__init__",
            [
              "REF",
              4
            ]
          ],
          [
            "__missing__",
            [
              "REF",
              5
            ]
          ],
          [
            "__reduce__",
            [
              "REF",
              6
            ]
          ],
          [
            "__repr__",
            [
              "REF",
              7
            ]
          ],
          [
            "copy",
            [
              "REF",
              8
            ]
          ],
          [
            "default_factory",
            [
              "REF",
              9
            ]
          ]
        ],
        "2": [
          "INSTANCE",
          "method_descriptor"
        ],
        "3": [
          "INSTANCE",
          "wrapper_descriptor"
        ],
        "4": [
          "INSTANCE",
          "wrapper_descriptor"
        ],
        "5": [
          "INSTANCE",
          "method_descriptor"
        ],
        "6": [
          "INSTANCE",
          "method_descriptor"
        ],
        "7": [
          "INSTANCE",
          "wrapper_descriptor"
        ],
        "8": [
          "INSTANCE",
          "method_descriptor"
        ],
        "9": [
          "INSTANCE",
          "member_descriptor"
        ],
        "10": [
          "INSTANCE",
          "defaultdict"
        ]
      },
      "stdout": ""
    },
    {
      "line": 4,
      "event": "step_line",
      "func_name": "<module>",
      "globals": {
        "defaultdict": [
          "REF",
          1
        ],
        "d": [
          "REF",
          10
        ]
      },
      "ordered_globals": [
        "defaultdict",
        "d"
      ],
      "stack_to_render": [],
      "heap": {
        "1": [
          "CLASS",
          "defaultdict",
          [
            "dict"
          ],
          [
            "__copy__",
            [
              "REF",
              2
            ]
          ],
          [
            "__getattribute__",
            [
              "REF",
              3
            ]
          ],
          [
            "__init__",
            [
              "REF",
              4
            ]
          ],
          [
            "__missing__",
            [
              "REF",
              5
            ]
          ],
          [
            "__reduce__",
            [
              "REF",
              6
            ]
          ],
          [
            "__repr__",
            [
              "REF",
              7
            ]
          ],
          [
            "copy",
            [
              "REF",
              8
            ]
          ],
          [
            "default_factory",
            [
              "REF",
              9
            ]
          ]
        ],
        "2": [
          "INSTANCE",
          "method_descriptor"
        ],
        "3": [
          "INSTANCE",
          "wrapper_descriptor"
        ],
        "4": [
          "INSTANCE",
          "wrapper_descriptor"
        ],
        "5": [
          "INSTANCE",
          "method_descriptor"
        ],
        "6": [
          "INSTANCE",
          "method_descriptor"
        ],
        "7": [
          "INSTANCE",
          "wrapper_descriptor"
        ],
        "8": [
          "INSTANCE",
          "method_descriptor"
        ],
        "9": [
          "INSTANCE",
          "member_descriptor"
        ],
        "10": [
          "INSTANCE",
          "defaultdict"
        ]
      },
      "stdout": ""
    },
    {
      "line": 5,
      "event": "step_line",
      "func_name": "<module>",
      "globals": {
        "defaultdict": [
          "REF",
          1
        ],
        "d": [
          "REF",
          10
        ]
      },
      "ordered_globals": [
        "defaultdict",
        "d"
      ],
      "stack_to_render": [],
      "heap": {
        "1": [
          "CLASS",
          "defaultdict",
          [
            "dict"
          ],
          [
            "__copy__",
            [
              "REF",
              2
            ]
          ],
          [
            "__getattribute__",
            [
              "REF",
              3
            ]
          ],
          [
            "__init__",
            [
              "REF",
              4
            ]
          ],
          [
            "__missing__",
            [
              "REF",
              5
            ]
          ],
          [
            "__reduce__",
            [
              "REF",
              6
            ]
          ],
          [
            "__repr__",
            [
              "REF",
              7
            ]
          ],
          [
            "copy",
            [
              "REF",
              8
            ]
          ],
          [
            "default_factory",
            [
              "REF",
              9
            ]
          ]
        ],
        "2": [
          "INSTANCE",
          "method_descriptor"
        ],
        "3": [
          "INSTANCE",
          "wrapper_descriptor"
        ],
        "4": [
          "INSTANCE",
          "wrapper_descriptor"
        ],
        "5": [
          "INSTANCE",
          "method_descriptor"
        ],
        "6": [
          "INSTANCE",
          "method_descriptor"
        ],
        "7": [
          "INSTANCE",
          "wrapper_descriptor"
        ],
        "8": [
          "INSTANCE",
          "method_descriptor"
        ],
        "9": [
          "INSTANCE",
          "member_descriptor"
        ],
        "10": [
          "INSTANCE",
          "defaultdict"
        ]
      },
      "stdout": ""
    },
    {
      "line": 5,
      "event": "return",
      "func_name": "<module>",
      "globals": {
        "defaultdict": [
          "REF",
          1
        ],
        "d": [
          "REF",
          10
        ]
      },
      "ordered_globals": [
        "defaultdict",
        "d"
      ],
      "stack_to_render": [],
      "heap": {
        "1": [
          "CLASS",
          "defaultdict",
          [
            "dict"
          ],
          [
            "__copy__",
            [
              "REF",
              2
            ]
          ],
          [
            "__getattribute__",
            [
              "REF",
              3
            ]
          ],
          [
            "__init__",
            [
              "REF",
              4
            ]
          ],
          [
            "__missing__",
            [
              "REF",
              5
            ]
          ],
          [
            "__reduce__",
            [
              "REF",
              6
            ]
          ],
          [
            "__repr__",
            [
              "REF",
              7
            ]
          ],
          [
            "copy",
            [
              "REF",
              8
            ]
          ],
          [
            "default_factory",
            [
              "REF",
              9
            ]
          ]
        ],
        "2": [
          "INSTANCE",
          "method_descriptor"
        ],
        "3": [
          "INSTANCE",
          "wrapper_descriptor"
        ],
        "4": [
          "INSTANCE",
          "wrapper_descriptor"
        ],
        "5": [
          "INSTANCE",
          "method_descriptor"
        ],
        "6": [
          "INSTANCE",
          "method_descriptor"
        ],
        "7": [
          "INSTANCE",
          "wrapper_descriptor"
        ],
        "8": [
          "INSTANCE",
          "method_descriptor"
        ],
        "9": [
          "INSTANCE",
          "member_descriptor"
        ],
        "10": [
          "INSTANCE",
          "defaultdict"
        ]
      },
      "stdout": ""
    }
  ]
}
