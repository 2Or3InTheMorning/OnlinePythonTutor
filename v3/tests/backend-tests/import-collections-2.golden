{
  "code": "from collections import defaultdict\nd = defaultdict(list)\nd['hello'].append('a')\nd['hello'].append('b')\nd['hello'].append('c')\n", 
  "trace": [
    {
      "ordered_globals": [], 
      "stdout": "", 
      "func_name": "<module>", 
      "stack_to_render": [], 
      "globals": {}, 
      "heap": {}, 
      "line": 1, 
      "event": "step_line"
    }, 
    {
      "ordered_globals": [
        "defaultdict"
      ], 
      "stdout": "", 
      "func_name": "<module>", 
      "stack_to_render": [], 
      "globals": {
        "defaultdict": [
          "REF", 
          1
        ]
      }, 
      "heap": {
        "1": [
          "CLASS", 
          "defaultdict", 
          [
            "dict"
          ], 
          [
            "__copy__", 
            [
              "REF", 
              2
            ]
          ], 
          [
            "__getattribute__", 
            [
              "REF", 
              3
            ]
          ], 
          [
            "__init__", 
            [
              "REF", 
              4
            ]
          ], 
          [
            "__missing__", 
            [
              "REF", 
              5
            ]
          ], 
          [
            "__reduce__", 
            [
              "REF", 
              6
            ]
          ], 
          [
            "__repr__", 
            [
              "REF", 
              7
            ]
          ], 
          [
            "copy", 
            [
              "REF", 
              8
            ]
          ], 
          [
            "default_factory", 
            [
              "REF", 
              9
            ]
          ]
        ], 
        "2": [
          "method_descriptor", 
          "<method '__copy__' of 'collections.defaultdict' objects>"
        ], 
        "3": [
          "wrapper_descriptor", 
          "<slot wrapper '__getattribute__' of 'collections.defaultdict' objects>"
        ], 
        "4": [
          "wrapper_descriptor", 
          "<slot wrapper '__init__' of 'collections.defaultdict' objects>"
        ], 
        "5": [
          "method_descriptor", 
          "<method '__missing__' of 'collections.defaultdict' objects>"
        ], 
        "6": [
          "method_descriptor", 
          "<method '__reduce__' of 'collections.defaultdict' objects>"
        ], 
        "7": [
          "wrapper_descriptor", 
          "<slot wrapper '__repr__' of 'collections.defaultdict' objects>"
        ], 
        "8": [
          "method_descriptor", 
          "<method 'copy' of 'collections.defaultdict' objects>"
        ], 
        "9": [
          "member_descriptor", 
          "<member 'default_factory' of 'collections.defaultdict' objects>"
        ]
      }, 
      "line": 2, 
      "event": "step_line"
    }, 
    {
      "ordered_globals": [
        "defaultdict", 
        "d"
      ], 
      "stdout": "", 
      "func_name": "<module>", 
      "stack_to_render": [], 
      "globals": {
        "defaultdict": [
          "REF", 
          1
        ], 
        "d": [
          "REF", 
          10
        ]
      }, 
      "heap": {
        "1": [
          "CLASS", 
          "defaultdict", 
          [
            "dict"
          ], 
          [
            "__copy__", 
            [
              "REF", 
              2
            ]
          ], 
          [
            "__getattribute__", 
            [
              "REF", 
              3
            ]
          ], 
          [
            "__init__", 
            [
              "REF", 
              4
            ]
          ], 
          [
            "__missing__", 
            [
              "REF", 
              5
            ]
          ], 
          [
            "__reduce__", 
            [
              "REF", 
              6
            ]
          ], 
          [
            "__repr__", 
            [
              "REF", 
              7
            ]
          ], 
          [
            "copy", 
            [
              "REF", 
              8
            ]
          ], 
          [
            "default_factory", 
            [
              "REF", 
              9
            ]
          ]
        ], 
        "2": [
          "method_descriptor", 
          "<method '__copy__' of 'collections.defaultdict' objects>"
        ], 
        "3": [
          "wrapper_descriptor", 
          "<slot wrapper '__getattribute__' of 'collections.defaultdict' objects>"
        ], 
        "4": [
          "wrapper_descriptor", 
          "<slot wrapper '__init__' of 'collections.defaultdict' objects>"
        ], 
        "5": [
          "method_descriptor", 
          "<method '__missing__' of 'collections.defaultdict' objects>"
        ], 
        "6": [
          "method_descriptor", 
          "<method '__reduce__' of 'collections.defaultdict' objects>"
        ], 
        "7": [
          "wrapper_descriptor", 
          "<slot wrapper '__repr__' of 'collections.defaultdict' objects>"
        ], 
        "8": [
          "method_descriptor", 
          "<method 'copy' of 'collections.defaultdict' objects>"
        ], 
        "9": [
          "member_descriptor", 
          "<member 'default_factory' of 'collections.defaultdict' objects>"
        ], 
        "10": [
          "collections.defaultdict", 
          "defaultdict(<type 'list'>, {})"
        ]
      }, 
      "line": 3, 
      "event": "step_line"
    }, 
    {
      "ordered_globals": [
        "defaultdict", 
        "d"
      ], 
      "stdout": "", 
      "func_name": "<module>", 
      "stack_to_render": [], 
      "globals": {
        "defaultdict": [
          "REF", 
          1
        ], 
        "d": [
          "REF", 
          10
        ]
      }, 
      "heap": {
        "1": [
          "CLASS", 
          "defaultdict", 
          [
            "dict"
          ], 
          [
            "__copy__", 
            [
              "REF", 
              2
            ]
          ], 
          [
            "__getattribute__", 
            [
              "REF", 
              3
            ]
          ], 
          [
            "__init__", 
            [
              "REF", 
              4
            ]
          ], 
          [
            "__missing__", 
            [
              "REF", 
              5
            ]
          ], 
          [
            "__reduce__", 
            [
              "REF", 
              6
            ]
          ], 
          [
            "__repr__", 
            [
              "REF", 
              7
            ]
          ], 
          [
            "copy", 
            [
              "REF", 
              8
            ]
          ], 
          [
            "default_factory", 
            [
              "REF", 
              9
            ]
          ]
        ], 
        "2": [
          "method_descriptor", 
          "<method '__copy__' of 'collections.defaultdict' objects>"
        ], 
        "3": [
          "wrapper_descriptor", 
          "<slot wrapper '__getattribute__' of 'collections.defaultdict' objects>"
        ], 
        "4": [
          "wrapper_descriptor", 
          "<slot wrapper '__init__' of 'collections.defaultdict' objects>"
        ], 
        "5": [
          "method_descriptor", 
          "<method '__missing__' of 'collections.defaultdict' objects>"
        ], 
        "6": [
          "method_descriptor", 
          "<method '__reduce__' of 'collections.defaultdict' objects>"
        ], 
        "7": [
          "wrapper_descriptor", 
          "<slot wrapper '__repr__' of 'collections.defaultdict' objects>"
        ], 
        "8": [
          "method_descriptor", 
          "<method 'copy' of 'collections.defaultdict' objects>"
        ], 
        "9": [
          "member_descriptor", 
          "<member 'default_factory' of 'collections.defaultdict' objects>"
        ], 
        "10": [
          "collections.defaultdict", 
          "defaultdict(<type 'list'>, {'hello': ['a']})"
        ]
      }, 
      "line": 4, 
      "event": "step_line"
    }, 
    {
      "ordered_globals": [
        "defaultdict", 
        "d"
      ], 
      "stdout": "", 
      "func_name": "<module>", 
      "stack_to_render": [], 
      "globals": {
        "defaultdict": [
          "REF", 
          1
        ], 
        "d": [
          "REF", 
          10
        ]
      }, 
      "heap": {
        "1": [
          "CLASS", 
          "defaultdict", 
          [
            "dict"
          ], 
          [
            "__copy__", 
            [
              "REF", 
              2
            ]
          ], 
          [
            "__getattribute__", 
            [
              "REF", 
              3
            ]
          ], 
          [
            "__init__", 
            [
              "REF", 
              4
            ]
          ], 
          [
            "__missing__", 
            [
              "REF", 
              5
            ]
          ], 
          [
            "__reduce__", 
            [
              "REF", 
              6
            ]
          ], 
          [
            "__repr__", 
            [
              "REF", 
              7
            ]
          ], 
          [
            "copy", 
            [
              "REF", 
              8
            ]
          ], 
          [
            "default_factory", 
            [
              "REF", 
              9
            ]
          ]
        ], 
        "2": [
          "method_descriptor", 
          "<method '__copy__' of 'collections.defaultdict' objects>"
        ], 
        "3": [
          "wrapper_descriptor", 
          "<slot wrapper '__getattribute__' of 'collections.defaultdict' objects>"
        ], 
        "4": [
          "wrapper_descriptor", 
          "<slot wrapper '__init__' of 'collections.defaultdict' objects>"
        ], 
        "5": [
          "method_descriptor", 
          "<method '__missing__' of 'collections.defaultdict' objects>"
        ], 
        "6": [
          "method_descriptor", 
          "<method '__reduce__' of 'collections.defaultdict' objects>"
        ], 
        "7": [
          "wrapper_descriptor", 
          "<slot wrapper '__repr__' of 'collections.defaultdict' objects>"
        ], 
        "8": [
          "method_descriptor", 
          "<method 'copy' of 'collections.defaultdict' objects>"
        ], 
        "9": [
          "member_descriptor", 
          "<member 'default_factory' of 'collections.defaultdict' objects>"
        ], 
        "10": [
          "collections.defaultdict", 
          "defaultdict(<type 'list'>, {'hello': ['a', 'b']})"
        ]
      }, 
      "line": 5, 
      "event": "step_line"
    }, 
    {
      "ordered_globals": [
        "defaultdict", 
        "d"
      ], 
      "stdout": "", 
      "func_name": "<module>", 
      "stack_to_render": [], 
      "globals": {
        "defaultdict": [
          "REF", 
          1
        ], 
        "d": [
          "REF", 
          10
        ]
      }, 
      "heap": {
        "1": [
          "CLASS", 
          "defaultdict", 
          [
            "dict"
          ], 
          [
            "__copy__", 
            [
              "REF", 
              2
            ]
          ], 
          [
            "__getattribute__", 
            [
              "REF", 
              3
            ]
          ], 
          [
            "__init__", 
            [
              "REF", 
              4
            ]
          ], 
          [
            "__missing__", 
            [
              "REF", 
              5
            ]
          ], 
          [
            "__reduce__", 
            [
              "REF", 
              6
            ]
          ], 
          [
            "__repr__", 
            [
              "REF", 
              7
            ]
          ], 
          [
            "copy", 
            [
              "REF", 
              8
            ]
          ], 
          [
            "default_factory", 
            [
              "REF", 
              9
            ]
          ]
        ], 
        "2": [
          "method_descriptor", 
          "<method '__copy__' of 'collections.defaultdict' objects>"
        ], 
        "3": [
          "wrapper_descriptor", 
          "<slot wrapper '__getattribute__' of 'collections.defaultdict' objects>"
        ], 
        "4": [
          "wrapper_descriptor", 
          "<slot wrapper '__init__' of 'collections.defaultdict' objects>"
        ], 
        "5": [
          "method_descriptor", 
          "<method '__missing__' of 'collections.defaultdict' objects>"
        ], 
        "6": [
          "method_descriptor", 
          "<method '__reduce__' of 'collections.defaultdict' objects>"
        ], 
        "7": [
          "wrapper_descriptor", 
          "<slot wrapper '__repr__' of 'collections.defaultdict' objects>"
        ], 
        "8": [
          "method_descriptor", 
          "<method 'copy' of 'collections.defaultdict' objects>"
        ], 
        "9": [
          "member_descriptor", 
          "<member 'default_factory' of 'collections.defaultdict' objects>"
        ], 
        "10": [
          "collections.defaultdict", 
          "defaultdict(<type 'list'>, {'hello': ['a', 'b', 'c']})"
        ]
      }, 
      "line": 5, 
      "event": "return"
    }
  ]
}
